tests/kernel/mem_protect/mem_protect/src/common.c

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2017 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <ztest.h>
+
+#define SYNC_SEMAPHORE_INIT_COUNT (0)
+#define SYNC_SEMAPHORE_MAX_COUNT (1)
+
+K_SEM_DEFINE(sync_sem, SYNC_SEMAPHORE_INIT_COUNT, SYNC_SEMAPHORE_MAX_COUNT);
+
+bool valid_fault;
```

Comment:
I had to introduce the use of a barrier (in the form of a system call for user mode threads) to ensure that this was always updated before the faulting instruction, see https://github.com/zephyrproject-rtos/zephyr/pull/5962.

Reply:
makes sense I will add the barriers made out of k_sem_give